### PR TITLE
edcas-client: init at 0.2.6

### PR DESCRIPTION
**Pull request by pabloaul**
## Description of changes
edcas-client is an information dashboard for the game Elite Dangerous

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ed/edcas-client/package.nix [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    libX11
+    wayland
+  ];
+in rustPlatform.buildRustPackage rec {
+  pname = "edcas-client";
+  version = "0.2.6";
+
+  src = fetchFromGitHub {
+    owner = "arne-fuchs";
+    repo = pname;
```
Comment (GetPsyched) [inline pkgs/by-name/ed/edcas-client/package.nix:30]: `repo` should not depend on `pname`
```suggestion
    repo = "edcas-client";
```

File: pkgs/by-name/ed/edcas-client/package.nix [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  rpathLibs = [
+    libglvnd
+    libXcursor
+    libXi
+    libxkbcommon
+    libXrandr
+    libX11
+    wayland
+  ];
+in rustPlatform.buildRustPackage rec {
```
Comment (GetPsyched) [inline pkgs/by-name/ed/edcas-client/package.nix:24]:
```suggestion
in
rustPlatform.buildRustPackage rec {
```

File: pkgs/by-name/ed/edcas-client/package.nix [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  src = fetchFromGitHub {
+    owner = "arne-fuchs";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-1IkaulUeTdAbLSaVmINaAMrfAK5SSGFpW2l04PPZylA=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = ./Cargo.lock;
+  };
```
Comment (GetPsyched) [inline pkgs/by-name/ed/edcas-client/package.nix:37]: Was reading the docs on this, isn't this only done when building a non-Rust package including Rust code?
Ref: https://nixos.org/manual/nixpkgs/stable/#compiling-rust-applications-with-cargo

File: pkgs/by-name/ed/edcas-client/package.nix [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    rustPlatform.bindgenHook
+  ];
+
+  postPatch = ''
+    substituteInPlace ./src/app/materials.rs --replace '/usr/share/edcas-client/' "$out/usr/share/"
```
Comment (GetPsyched) [inline pkgs/by-name/ed/edcas-client/package.nix:55]:
```suggestion
    substituteInPlace ./src/app/materials.rs --replace '/usr/share/edcas-client/' "$out/share/"
```

File: pkgs/by-name/ed/edcas-client/package.nix [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    substituteInPlace ./src/app/settings.rs --replace '/etc/edcas-client/' "$out/etc/"
+    substituteInPlace ./src/app/about.rs --replace '/usr/share/edcas-client/' "$out/"
+    substituteInPlace ./src/app/news.rs --replace '/usr/share/edcas-client/' "$out/"
+    substituteInPlace ./src/main.rs --replace '/usr/share/edcas-client/' "$out/"
```
Comment (GetPsyched) [inline pkgs/by-name/ed/edcas-client/package.nix:59]: Typo?
```suggestion
    substituteInPlace ./src/app/about.rs --replace '/usr/share/edcas-client/' "$out/share/"
    substituteInPlace ./src/app/news.rs --replace '/usr/share/edcas-client/' "$out/share/"
    substituteInPlace ./src/main.rs --replace '/usr/share/edcas-client/' "$out/share/"
```

File: pkgs/by-name/ed/edcas-client/package.nix [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+  nativeBuildInputs = [
+    pkg-config
+    cmake
+    rustPlatform.bindgenHook
+  ];
+
+  postPatch = ''
+    substituteInPlace ./src/app/materials.rs --replace '/usr/share/edcas-client/' "$out/usr/share/"
+    substituteInPlace ./src/app/settings.rs --replace '/etc/edcas-client/' "$out/etc/"
```
Comment (GetPsyched) [inline pkgs/by-name/ed/edcas-client/package.nix:56]: This is likely not required. Doubtful though, should wait for another reviewer.

File: pkgs/by-name/ed/edcas-client/package.nix [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    substituteInPlace ./src/main.rs --replace '/usr/share/edcas-client/' "$out/"
+  '';
+
+  postFixup = ''
+    patchelf --set-rpath "${lib.makeLibraryPath rpathLibs}:$(patchelf --print-rpath $out/bin/edcas-client)" $out/bin/edcas-client
+  '';
+
+  postInstall = ''
+    mkdir -p $out/etc/ $out/usr/share $out/graphics/logo
+    install -Dm644 $src/graphics/logo/edcas_128.png $src/graphics/logo/edcas.png "$out/graphics/logo"
```
Comment (GetPsyched) [inline pkgs/by-name/ed/edcas-client/package.nix:68]:
```suggestion
    install -D -m 644 $src/graphics/logo/edcas_128.png $out/share/icons/hicolor/128x128/apps/edcas.png
```
Also, consider adding a `.desktop` file (using `copyDesktopItems` and `makeDesktopItem`)

**Review by GetPsyched — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ed/edcas-client/package.nix (hunk 6 above)
Comment (pabloaul) [reply pkgs/by-name/ed/edcas-client/package.nix:57]: The upstream package does the path handling manually so they have to be patched for the program to not crash.

**Comment by GetPsyched** [comment]
You'd want to amend to the older init commit instead of creating a new one. Please see the [commit guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)

**Comment by pabloaul** [comment]
> You'd want to amend to the older init commit instead of creating a new one. Please see the [commit guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions)

If I understand this correctly, the commit tagged with fixup should be squashed automatically so this should be okay.
I am still fixing the substitution paths because the missing icons now cause it to crash...

**Comment by GetPsyched** [comment]
BTW, you didn't remove the `Cargo.lock` file from the commit (since you removed the import)

**Comment by pabloaul** [comment]
> BTW, you didn't remove the `Cargo.lock` file from the commit (since you removed the import)

Yes, I was able to remove the explicit Cargo.lock import and have it still fetch it.
However the CI here fails without the file so I added it back. If you have a better idea to make this work, please tell me.

Also thank a lot for looking over this PR!

**Comment by GetPsyched** [comment]
> If you have a better idea to make this work, please tell me.

Sorry, but I don't. I haven't worked around Rust packaging a lot and whatever I reviewed was based on the docs. I think I'll leave this here for another reviewer to pick up :)

**Comment by pabloaul** [comment]
will open another PR when upstream fixes the path handling

**Pull request closed without merging**